src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -0,0 +1,536 @@
+
+        internal static float SumAbsU(float[] src, int count)
+        {
+            Vector128<float> result = Sse.SetZeroVector128();
+            Vector128<float> mask;
+
+            if (Sse2.IsSupported)
+            {
+                mask = Sse.StaticCast<int, float>(Sse2.SetAllVector128(0x7FFFFFFF));
+            }
+            else
+            {
+                mask = Sse.SetAllVector128(0x7FFFFFFF);
+            }
+
+            unsafe
+            {
+                fixed (float* psrc = src)
+                {
+                    float* pSrcCurrent = psrc;
+                    float* pEnd = psrc + count;
+
+                    while (pSrcCurrent + 4 <= pEnd)
+                    {
+                        Vector128<float> srcVector = Sse.LoadVector128(pSrcCurrent);
+                        result = Sse.Add(result, Sse.And(srcVector, mask));
+
+                        pSrcCurrent += 4;
+                    }
+
+                    // TODO: Check code length per line
+                    if (Sse3.IsSupported)
+                    {
+                        // SSE3 is supported.
+                        result = Sse3.HorizontalAdd(result, result);
+                        result = Sse3.HorizontalAdd(result, result);
+                    }
+                    else
+                    {
+                        // SSE3 is not supported.
+                        result = Sse.Add(result, Sse.MoveHighToLow(result, result));
+                        result = Sse.Add(result, Sse.MoveHighToLow(result, Sse.UnpackLow(result, result)));
+                    }
+
+                    while (pSrcCurrent < pEnd)
+                    {
+                        Vector128<float> srcVector = Sse.LoadVector128(pSrcCurrent);
+                        result = Sse.Add(result, Sse.And(srcVector, mask));
+
+                        pSrcCurrent++;
+                    }
+                }
+            }
+
+            return Sse.ConvertToSingle(result);
+        }
+
+        internal static float DotU(float[] src, float[] dst, int count)
+        {
+            Vector128<float> result = Sse.SetZeroVector128();
+
+            unsafe
+            {
+
+                fixed (float* psrc = src)
+                fixed (float* pdst = dst)
+                {
+                    float* pSrcCurrent = psrc;
+                    float* pDstCurrent = pdst;
+                    float* pEnd = psrc + count;
+
+                    while (pSrcCurrent + 4 <= pEnd)
+                    {
+                        Vector128<float> srcVector = Sse.LoadVector128(pSrcCurrent);
+                        Vector128<float> dstVector = Sse.LoadVector128(pDstCurrent);
+
+                        result = Sse.Add(result, Sse.Multiply(srcVector, dstVector));
+
+                        pSrcCurrent += 4;
+                        pDstCurrent += 4;
+                    }
+
+                    if (Sse3.IsSupported)
+                    {
+                        // SSE3 is supported.
```

Comment:
I don't think this comment adds much (it literally restates the statement immediately proceeding) so I would remove these.

`// SEE3 is not supported` is OK because it saves you having to read upwards to find the conditional clauses.